KVM live storage migration intra cluster from NFS source and destination

nvazquez opened the pull request:
## Description

Feature Specification: https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=95653548

Live storage migration on KVM under these conditions:
- From source and destination hosts within the same cluster
- From NFS primary storage to NFS cluster-wide primary storage
- Source NFS and destination NFS storage mounted on hosts

In order to enable this functionality, database should be updated in order to enable live storage capacibilty for KVM, if previous conditions are met. This is due to existing conflicts between qemu and libvirt versions. This has been tested on CentOS 6 hosts.

Additional notes:

1. To use this feature set the `storage_motion_supported=1` in the hypervisor_capability table for KVM. This is done by default as the feature may not work in some environments, read below.
2. This feature of online storage+VM migration for KVM will only work with CentOS6 and possible Ubuntu as KVM hosts but not with CentOS7 due to:
- https://bugs.centos.org/view.php?id=14026
- https://bugzilla.redhat.com/show_bug.cgi?id=1219541
On CentOS7 the error we see is: `" error: unable to execute QEMU command 'migrate': this feature or command is not currently supported"` (reference https://ask.openstack.org/en/question/94186/live-migration-unable-to-execute-qemu-command-migrate/). Reading through various lists looks like the migrate feature with qemu may be available with paid versions of RHEL-EV but not centos7 however this works with CentOS6.

Fix for CentOS 7:
- Create repo file on /etc/yum.repos.d/:
````
[qemu-kvm-rhev]
name=oVirt rebuilds of qemu-kvm-rhev
baseurl=http://resources.ovirt.org/pub/ovirt-3.5/rpm/el7Server/
mirrorlist=http://resources.ovirt.org/pub/yum-repo/mirrorlist-ovirt-3.5-el7Server
enabled=1
skip_if_unavailable=1
gpgcheck=0
````
- `yum install qemu-kvm-common-ev-2.3.0-29.1.el7.x86_64 qemu-kvm-ev-2.3.0-29.1.el7.x86_64 qemu-img-ev-2.3.0-29.1.el7.x86_64`
- Reboot host

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Screenshots (if appropriate):

## How Has This Been Tested?
Tested on environment having 2 KVM CentOS 6 hosts.

Note: CentOS 7 hosts report an issue around qemu version:
`org.libvirt.LibvirtException: internal error: unable to execute QEMU command 'migrate': this feature or command is not currently supported`

Fix for CentOS 7:
- Create repo file on /etc/yum.repos.d/:
````
[qemu-kvm-rhev]
name=oVirt rebuilds of qemu-kvm-rhev
baseurl=http://resources.ovirt.org/pub/ovirt-3.5/rpm/el7Server/
mirrorlist=http://resources.ovirt.org/pub/yum-repo/mirrorlist-ovirt-3.5-el7Server
enabled=1
skip_if_unavailable=1
gpgcheck=0
````
- `yum install qemu-kvm-common-ev-2.3.0-29.1.el7.x86_64 qemu-kvm-ev-2.3.0-29.1.el7.x86_64 qemu-img-ev-2.3.0-29.1.el7.x86_64`
- Reboot host

DaanHoogland commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -250,6 +264,52 @@ private boolean canHandle(DataObject dataObject) {
+     * - Source and destination storage are NFS
+     * - Destination storage is cluster-wide
+     */
+    protected StrategyPriority canHandleKVMNonManagedLiveStorageMigration(Map<VolumeInfo, DataStore> volumeMap,
```
I am missing the check for hypervisor type == KVM, am I?

DaanHoogland replied:
never mind, happens up the stack

DaanHoogland commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -1675,6 +1737,64 @@ private SnapshotDetailsVO handleSnapshotDetails(long csSnapshotId, String value)
         return _snapshotDetailsDao.persist(snapshotDetails);
     }
 
+    /**
```
this file is growing over 300 lines. Can we move the KVM live migration code to a TacticalKVMMigrationAid class? (name just a suggestion)

DaanHoogland commented on plugins/hypervisors/kvm/src/test/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapperTest.java:
```diff
@@ -325,4 +442,27 @@ public void testMigrationUriException() {
+        migrateStorage.put(disk1SourceFilename, migrateDisk1Info);
+        migrateStorage.put(disk2SourceFilename, migrateDisk2Info);
+        String newXml = lw.replaceStorage(sourceMultidiskDomainXml, migrateStorage, false);
+        assertTrue(newXml.contains(destDisk1Path));
```
rather not check for the exact path? (like the assertFalse checks below)

nvazquez replied:
It is actually checking the full path as it is being defined as: `String destDisk1Path = String.format("/mnt/%s/%s", destPoolUuid, destDisk1FileName)`

wido commented on core/src/main/java/com/cloud/agent/api/storage/CheckStorageAvailabilityCommand.java:
```diff
@@ -0,0 +1,43 @@
+    public boolean executeInSequence() {
+        return false;
+    }
+}
```
Add a newline here :)

nvazquez replied:
Done, thanks

wido commented on engine/schema/src/main/resources/META-INF/db/schema-41120to41200.sql:
```diff
@@ -37,4 +37,4 @@ INSERT INTO `cloud`.`role_permissions` (`uuid`, `role_id`, `rule`, `permission`,
 -- PR#1448 update description of 'execute.in.sequence.network.element.commands' parameter to reflect an unused command that has been removed. The removed class command is 'UserDataCommand'.
-update `cloud`.`configuration` set description = 'If set to true, DhcpEntryCommand, SavePasswordCommand, VmDataCommand will be synchronized on the agent side. If set to false, these commands become asynchronous. Default value is false.' where name = 'execute.in.sequence.network.element.commands'; 
\ No newline at end of file
+update `cloud`.`configuration` set description = 'If set to true, DhcpEntryCommand, SavePasswordCommand, VmDataCommand will be synchronized on the agent side. If set to false, these commands become asynchronous. Default value is false.' where name = 'execute.in.sequence.network.element.commands';
```
Update by accident of this file?

nvazquez replied:
Had previously included an update sql on the `hypervisor_cappabilities` to enable live storage motion on KVM. But due to conflicting versions on qemu just decided to remove it and must have removed an end of line or extra space, will sort it out

wido commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java:
```diff
@@ -30,8 +31,13 @@
 
+    private String getQemuImgPathScript = String.format("which %s >& /dev/null; " +
+                    "if [ $? -gt 0 ]; then echo \"%s\"; else echo \"%s\"; fi",
+            cloudQemuImgPath, _qemuImgPath, cloudQemuImgPath);
```
We have a Qemu utility in the KVM code, can't we offload this there?

nvazquez replied:
As this feature requires qemu-img command supporting snapshots I have introduced this check. This cloud-qemu-img can be included on KVM hosts to support snapshots as per: http://www.nux.ro/oldblog/archive/2014/01/Taking_KVM_volume_snapshots_with_Cloudstack_4_2_on_CentOS_6_5.html
If cloud-qemu-img is present then we can use it, and if not just qemu-img. It is also done this way on scripts: https://github.com/apache/cloudstack/search?q=cloud-qemu-img&unscoped_q=cloud-qemu-img.

wido commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java:
```diff
@@ -30,8 +31,13 @@
 
     /* The qemu-img binary. We expect this to be in $PATH */
     public String _qemuImgPath = "qemu-img";
+    private String cloudQemuImgPath = "cloud-qemu-img";
```
This binary does not exist on Ubuntu/Debian. Why the name change?

nvazquez replied:
Just to make sure it works with snapshot support. If the binary is present on the KVM host it will be used, and if not just the qemu-img.

nvazquez commented:
Thanks @wido @DaanHoogland for reviewing, I'll address your comments soon. Along with @borisstoyanov we'll be adding integration tests on this PR.

nvazquez commented:
Feature specification link added to the PR description

mike-tutkowski commented:
@nvazquez Hi Nicolas - Just an FYI that I plan to look at this tomorrow.

nvazquez commented:
Thanks @mike-tutkowski that's great

mike-tutkowski commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -1991,8 +2198,49 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
                 throw new CloudRuntimeException("Destination storage pool with ID " + dataStore.getId() + " was not located.");
             }
 
+            if (storageTypeConsistency == null) {
```
Won't this expression always evaluate to true?

nvazquez replied:
Actually just for the first item in the map entry set, as the variable is defined out of the `for` block. The idea is to set it on the first iteration and compare it on the next ones.

mike-tutkowski replied:
Ah, yes, my mistake. The code view had a section collapsed and I didn't see the looping.

OK, I've gone through the contents of the PR and it LGTM. Thanks!

rhtyd commented:
@nvazquez I've rebased and fixed conflicts against latest master. Can you check the final rebased commit against the original commit here: https://github.com/apache/cloudstack/pull/2983/commits/5439cf39509566f4ae98bb69afc7899f253f6565

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2509

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@blueorangutan test matrix

blueorangutan commented:
@rhtyd a Trillian-Jenkins matrix job (centos6 mgmt + xs71, centos7 mgmt + vmware65, centos7 mgmt + kvmcentos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3278)</b>
Environment: xenserver-71 (x2), Advanced Networking with Mgmt server 7
Total time taken: 15866 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3278-xenserver-71.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermittent failure detected: /marvin/tests/smoke/test_reset_vm_on_reboot.py
Intermittent failure detected: /marvin/tests/smoke/test_resource_accounting.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_secondary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_templates.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Smoke tests completed. 47 look OK, 23 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 77.67 | test_multipleips_per_nic.py
test_02_vpc_privategw_static_routes | `Failure` | 206.87 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 367.92 | test_privategw_acl.py
ContextSuite context=TestResetVmOnReboot>:setup | `Error` | 0.00 | test_reset_vm_on_reboot.py
test_01_so_removal_resource_update | `Error` | 1.46 | test_resource_accounting.py
ContextSuite context=TestRouterDHCPHosts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPOpts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDns>:setup | `Error` | 0.00 | test_router_dns.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestLBRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestNatRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestPublicIPUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestSnapshotUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVmUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVolumeUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVpnUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestRouterDnsService>:setup | `Error` | 0.00 | test_router_dnsservice.py
test_02_routervm_iptables_policies | `Error` | 1.36 | test_routers_iptables_default_policy.py
test_01_single_VPC_iptables_policies | `Error` | 3.37 | test_routers_iptables_default_policy.py
test_01_isolate_network_FW_PF_default_routes_egress_true | `Error` | 1.50 | test_routers_network_ops.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Error` | 1.43 | test_routers_network_ops.py
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Error` | 6.24 | test_routers_network_ops.py
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Error` | 5.96 | test_routers_network_ops.py
test_03_RVR_Network_check_router_state | `Error` | 6.01 | test_routers_network_ops.py
ContextSuite context=TestRouterServices>:setup | `Error` | 0.00 | test_routers.py
ContextSuite context=TestScaleVm>:setup | `Error` | 0.00 | test_scale_vm.py
test_01_sys_vm_start | `Failure` | 0.14 | test_secondary_storage.py
ContextSuite context=TestServiceOfferings>:setup | `Error` | 2.46 | test_service_offerings.py
ContextSuite context=TestCreateVolume>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestVolumes>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestSnapshotRootDisk>:setup | `Error` | 0.00 | test_snapshots.py
test_01_list_sec_storage_vm | `Failure` | 0.05 | test_ssvm.py
test_02_list_cpvm_vm | `Failure` | 0.05 | test_ssvm.py
test_03_ssvm_internals | `Failure` | 0.06 | test_ssvm.py
test_04_cpvm_internals | `Failure` | 0.06 | test_ssvm.py
test_05_stop_ssvm | `Failure` | 0.06 | test_ssvm.py
test_06_stop_cpvm | `Failure` | 0.05 | test_ssvm.py
test_07_reboot_ssvm | `Failure` | 0.06 | test_ssvm.py
test_08_reboot_cpvm | `Failure` | 0.06 | test_ssvm.py
test_09_destroy_ssvm | `Failure` | 0.06 | test_ssvm.py
test_10_destroy_cpvm | `Failure` | 0.06 | test_ssvm.py
test_02_create_template_with_checksum_sha1 | `Error` | 65.57 | test_templates.py
test_03_create_template_with_checksum_sha256 | `Error` | 65.59 | test_templates.py
test_04_create_template_with_checksum_md5 | `Error` | 65.59 | test_templates.py
test_05_create_template_with_no_checksum | `Error` | 65.58 | test_templates.py
ContextSuite context=TestTemplates>:setup | `Error` | 3.11 | test_templates.py
ContextSuite context=Test01DeployVM>:setup | `Error` | 0.00 | test_vm_life_cycle.py
ContextSuite context=Test02VMLifeCycle>:setup | `Error` | 0.00 | test_vm_life_cycle.py
test_change_service_offering_for_vm_with_snapshots | `Error` | 1.37 | test_vm_snapshots.py
ContextSuite context=TestVmSnapshot>:setup | `Error` | 11.09 | test_vm_snapshots.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 7.29 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 6.28 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 8.32 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 7.26 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 6.25 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Error` | 5.25 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Error` | 6.22 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Failure` | 6.56 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Failure` | 5.49 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | `Failure` | 3.27 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Failure` | 3.43 | test_vpc_vpn.py

blueorangutan commented:
<b>Trillian test result (tid-3280)</b>
Environment: vmware-65 (x2), Advanced Networking with Mgmt server 7
Total time taken: 16636 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3280-vmware-65.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermittent failure detected: /marvin/tests/smoke/test_reset_vm_on_reboot.py
Intermittent failure detected: /marvin/tests/smoke/test_resource_accounting.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_secondary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_templates.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Smoke tests completed. 48 look OK, 22 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 89.64 | test_multipleips_per_nic.py
ContextSuite context=TestResetVmOnReboot>:setup | `Error` | 0.00 | test_reset_vm_on_reboot.py
test_01_so_removal_resource_update | `Error` | 1.27 | test_resource_accounting.py
ContextSuite context=TestRouterDHCPHosts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPOpts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDns>:setup | `Error` | 0.00 | test_router_dns.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestLBRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestNatRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestPublicIPUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestSnapshotUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVmUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVolumeUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVpnUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestRouterDnsService>:setup | `Error` | 0.00 | test_router_dnsservice.py
test_02_routervm_iptables_policies | `Error` | 1.20 | test_routers_iptables_default_policy.py
test_01_single_VPC_iptables_policies | `Error` | 4.27 | test_routers_iptables_default_policy.py
test_01_isolate_network_FW_PF_default_routes_egress_true | `Error` | 1.28 | test_routers_network_ops.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Error` | 1.27 | test_routers_network_ops.py
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Error` | 4.90 | test_routers_network_ops.py
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Error` | 5.99 | test_routers_network_ops.py
test_03_RVR_Network_check_router_state | `Error` | 6.00 | test_routers_network_ops.py
ContextSuite context=TestRouterServices>:setup | `Error` | 0.00 | test_routers.py
ContextSuite context=TestScaleVm>:setup | `Error` | 0.00 | test_scale_vm.py
test_01_sys_vm_start | `Failure` | 0.09 | test_secondary_storage.py
ContextSuite context=TestServiceOfferings>:setup | `Error` | 1.86 | test_service_offerings.py
ContextSuite context=TestCreateVolume>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestVolumes>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestSnapshotRootDisk>:setup | `Error` | 0.00 | test_snapshots.py
test_01_list_sec_storage_vm | `Failure` | 0.04 | test_ssvm.py
test_02_list_cpvm_vm | `Failure` | 0.03 | test_ssvm.py
test_03_ssvm_internals | `Failure` | 0.04 | test_ssvm.py
test_04_cpvm_internals | `Failure` | 0.04 | test_ssvm.py
test_05_stop_ssvm | `Failure` | 0.03 | test_ssvm.py
test_06_stop_cpvm | `Failure` | 0.03 | test_ssvm.py
test_07_reboot_ssvm | `Failure` | 0.03 | test_ssvm.py
test_08_reboot_cpvm | `Failure` | 0.04 | test_ssvm.py
test_09_destroy_ssvm | `Failure` | 0.03 | test_ssvm.py
test_10_destroy_cpvm | `Failure` | 0.03 | test_ssvm.py
test_02_create_template_with_checksum_sha1 | `Error` | 65.39 | test_templates.py
test_03_create_template_with_checksum_sha256 | `Error` | 65.41 | test_templates.py
test_04_create_template_with_checksum_md5 | `Error` | 65.42 | test_templates.py
test_05_create_template_with_no_checksum | `Error` | 65.40 | test_templates.py
ContextSuite context=TestTemplates>:setup | `Error` | 2.18 | test_templates.py
ContextSuite context=Test01DeployVM>:setup | `Error` | 0.00 | test_vm_life_cycle.py
ContextSuite context=Test02VMLifeCycle>:setup | `Error` | 0.00 | test_vm_life_cycle.py
test_change_service_offering_for_vm_with_snapshots | `Error` | 1.24 | test_vm_snapshots.py
ContextSuite context=TestVmSnapshot>:setup | `Error` | 6.22 | test_vm_snapshots.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 6.78 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 6.76 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 5.75 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 5.79 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 5.76 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Error` | 5.82 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Error` | 3.70 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Failure` | 7.58 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Failure` | 3.32 | test_vpc_vpn.py
test_01_vpc_remote_access_vpn | `Failure` | 4.20 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Failure` | 4.36 | test_vpc_vpn.py

rhtyd commented:
@blueorangutan test matrix

blueorangutan commented:
@rhtyd a Trillian-Jenkins matrix job (centos6 mgmt + xs71, centos7 mgmt + vmware65, centos7 mgmt + kvmcentos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3281)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 26982 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3281-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 68 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 48.22 | test_multipleips_per_nic.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 517.25 | test_vpc_redundant.py

blueorangutan commented:
<b>Trillian test result (tid-3282)</b>
Environment: xenserver-71 (x2), Advanced Networking with Mgmt server 7
Total time taken: 26828 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3282-xenserver-71.zip
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Smoke tests completed. 67 look OK, 3 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 56.39 | test_multipleips_per_nic.py
test_01_scale_vm | `Failure` | 16.53 | test_scale_vm.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 528.26 | test_vpc_redundant.py

blueorangutan commented:
<b>Trillian test result (tid-3286)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 24366 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3286-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 68 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_nic_secondaryip_add_remove | `Error` | 31.20 | test_multipleips_per_nic.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 496.74 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 0.48 | test_vpc_redundant.py

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2511

blueorangutan commented:
<b>Trillian test result (tid-3295)</b>
Environment: xenserver-71 (x2), Advanced Networking with Mgmt server 7
Total time taken: 30664 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3295-xenserver-71.zip
Intermittent failure detected: /marvin/tests/smoke/test_accounts.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups_projects.py
Intermittent failure detected: /marvin/tests/smoke/test_async_job.py
Intermittent failure detected: /marvin/tests/smoke/test_multipleips_per_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Smoke tests completed. 64 look OK, 6 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestAccounts>:setup | `Error` | 0.00 | test_accounts.py
ContextSuite context=TestAddVmToSubDomain>:setup | `Error` | 0.00 | test_accounts.py
test_DeleteDomain | `Error` | 1.75 | test_accounts.py
test_forceDeleteDomain | `Error` | 1.68 | test_accounts.py
ContextSuite context=TestRemoveUserFromAccount>:setup | `Error` | 6.05 | test_accounts.py
ContextSuite context=TestDeployVmWithAffinityGroup>:setup | `Error` | 0.00 | test_affinity_groups_projects.py
test_query_async_job_result | `Error` | 0.00 | test_async_job.py
test_nic_secondaryip_add_remove | `Error` | 65.06 | test_multipleips_per_nic.py
test_01_scale_vm | `Failure` | 22.75 | test_scale_vm.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 543.46 | test_vpc_redundant.py

rhtyd commented:
Thanks @GabrielBrascher I've also asked @borisstoyanov to test and review this.

I'll send the doc PR soon, meanwhile please NOTE:
1. To use this feature set the `storage_motion_supported=1` in the hypervisor_capability table for KVM. This is done by default as the feature may not work in some environments, read below.
2. This feature of online storage+VM migration for KVM will only work with CentOS6 and possible Ubuntu as KVM hosts but not with CentOS7 due to:
- https://bugs.centos.org/view.php?id=14026
- https://bugzilla.redhat.com/show_bug.cgi?id=1219541
On CentOS7 the error we see is: `" error: unable to execute QEMU command 'migrate': this feature or command is not currently supported"` (reference https://ask.openstack.org/en/question/94186/live-migration-unable-to-execute-qemu-command-migrate/). Reading through various lists looks like the migrate feature with qemu may be available with paid versions of RHEL-EV but not centos7 however this works with CentOS6.

@blueorangutan package

rhtyd commented:
@blueorangutan package

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
Test LGTM
@GabrielBrascher are you LGTM on this, or still reviewing?

rhtyd commented:
Found one issue where one global setting was not available on upgraded systems as it was declared in Config.java but not as a configkey. My above commit fixes that. /cc @borisstoyanov @DagSonsteboSB
https://github.com/apache/cloudstack/pull/2983/commits/73d6f955f14dd58710b8248b9739f1b38c433b85

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-2552

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2553

blueorangutan commented:
@GabrielBrascher a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3336)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 23041 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3336-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 70 look OK, 0 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---

GabrielBrascher commented:
@rhtyd the schedule is getting tight indeed, but I think that we can still merge this one. Reviews are looking good and overall the code is OK.

If tests pass (they are looking good) and we have this PR stable, I have no concerns in merging it on 4.12.0.0. Said that, could you please fix the conflicting file? Thanks!

The bigger issue that I see here (and also on my PRs #3154 and #2997) is the whole architecture as it is. As I mentioned in #3154, using the KvmNonManagedStorageDataMotionStrategy was the best that I could think regarding using the current architecture and easying future maintenance and extensions. Ideally, we should refactor this class structure and use proper O.O. However, this is not a simple task, and as such, I think it is best executed separated from any new feature introduction.

rhtyd commented:
@GabrielBrascher thanks, I'll get the conflict by tomorrow and kick a final round of tests.

sureshanaparti commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
 
+    /**
+     * True if volumes source storage are NFS
+     */
```
@nvazquez The below method returns true when the first NFS pool of a volume found in the given volumes list. Is that the intent? or it has to check all the volumes with NFS pool?

rhtyd replied:
Yes.

sureshanaparti replied:
@rhyd Ok.

nvazquez replied:
@sureshanaparti yes, that was the intent as it was planned to be for live storage migrations from host H1 to host H2 and storage A to storage B, i.e. multiple volumes on the same storage so no need to check all of them. But it can be extended to check all the volumes in the map

sureshanaparti replied:
@nvazquez If multiple volumes are on the same storage is what being checked, why Map with multiple volumes & looping around them is required?

it can be extended? any current use case supports this? if extended, it only checks for NFS pool.

sureshanaparti commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
+    /**
+     * True if destination storage is cluster-wide NFS
+     */
+    protected boolean isDestinationNfsPrimaryStorageClusterWide(Map<VolumeInfo, DataStore> volumeMap) {
```
@nvazquez The volume info in the Map is unused. Better pass pools list as input here.

rhtyd replied:
Same as before, the consumer will have to check the map for nullity, unbox the map and pass the list.

sureshanaparti replied:
Understood. Either case nullity check has to be done. But, I think this is not a good coding practice, to pass the unused data to the method.

rhtyd replied:
Well depends how you'd want to look at it. It's not unused if you're using the collection.

sureshanaparti replied:
For a datastore validation, only datastore details are required.

rhtyd commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
+     * - Source and destination storage are NFS
+     * - Destination storage is cluster-wide
+     */
+    protected StrategyPriority canHandleKVMNonManagedLiveStorageMigration(Map<VolumeInfo, DataStore> volumeMap,
```
I think the `KvmNonManagedStorageDataMotionStrategy` and `StorageSystemDataMotionStrategy` are both instantiated and consumed the same way, we'll have to check if by simply moving the changes will not break the API. I'll discuss with @nvazquez and get back on this, we can submit a refactoring PR in future.

nvazquez replied:
Agree with moving it to `KvmNonManagedStorageDataMotionStrategy` class

rhtyd commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
+    /**
+     * True if volumes source storage are NFS
+     */
+    protected boolean isSourceNfsPrimaryStorage(Map<VolumeInfo, DataStore> volumeMap) {
```
The consumer of the method will have to do the Map null check then.

rhtyd commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -253,18 +262,64 @@ private boolean canHandle(DataObject dataObject) {
+    }
+
+    /**
+     * True if destination storage is cluster-wide NFS
```
Yes.

nvazquez replied:
Same as above

sureshanaparti replied:
@nvazquez ok. this also can be extended for multiple datastore check?

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

sureshanaparti commented on api/src/main/java/com/cloud/storage/MigrationOptions.java:
```diff
@@ -0,0 +1,86 @@
+    private boolean copySrcTemplate;
+    private String srcVolumeUuid;
+    private int timeout;
+
```
The object state is different for Clone types: LinkedClone (**srcPoolUuid**, **srcPoolType**, _srcBackingFilePath_, _copySrcTemplate_) and FullClone (**srcPoolUuid**, **srcPoolType**, _srcVolumeUuid_). May be better class design here?

rhtyd replied:
Well the idea of the class is to capture migration options, I don't see why two constructors cannot be used.

sureshanaparti replied:
I mean State and Behavior of the class is not properly defined.

sureshanaparti replied:
At any given point of time, the state of the object is incomplete.

rhtyd replied:
There is no *behaviour* in this class, the purpose of the class is to hold data/config (or state as you will).

sureshanaparti replied:
yes, the same i wanted to convey here. Based on the types, the fields are not relevant. Can use only relevant fields with separate classes?

sureshanaparti commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -2046,6 +2208,51 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
+     */
+    private void verifyDestinationStorage(Map<String, Storage.StoragePoolType> sourcePools, Host destHost) {
+        if (MapUtils.isNotEmpty(sourcePools)) {
+            LOGGER.debug("Verifying NFS source pools are already mounted on destination host " + destHost.getUuid());
```
Above log message says `NFS source pools`. But, the storage pools with respective `Storage.StoragePoolType` indicate the other pool types can be a valid input. Please correct the log message if other pool types are sent for verification. Else pool type is not required.

rhtyd replied:
Fixed.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2580

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

GabrielBrascher commented:
@blueorangutan test

blueorangutan commented:
@GabrielBrascher a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@GabrielBrascher the test will most probably will fail due to a pending packaging job, kindly wait for packaging to come back before tests could be kicked.

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3360)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 25097 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3360-kvm-centos7.zip
Smoke tests completed. 70 look OK, 0 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---

GabrielBrascher commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -2022,10 +2153,40 @@ private ModifyTargetsCommand getModifyTargetsCommand(long storagePoolId, String
+    *                  the destination storages can be all managed or all not managed, not mixed.
     */
-    private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataStoreMap) {
+    protected void verifyLiveMigrationForKVM(Map<VolumeInfo, DataStore> volumeDataStoreMap, Host destHost) {
```
What about implementing such verification implemented on the canHandle?

nvazquez replied:
It could be. I think it does not harm to keep a detailed verification and get a proper error in case the requirements are not met. In the case of `canHandle` it would be harder to track as in case the requirements are not met then the strategy will not handle the migration and would need to log the issue. Both ways are valid but I decided the first one. What do you think? @GabrielBrascher @rhtyd

GabrielBrascher commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -2046,6 +2207,51 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
+
+            if (!destStoragePoolVO.isManaged()) {
+                if (srcStoragePoolVO.getPoolType() != Storage.StoragePoolType.NetworkFilesystem ||
+                        destStoragePoolVO.getPoolType() != Storage.StoragePoolType.NetworkFilesystem) {
```
This verification blocks the migration between local storage (storage type filesystem), allowed with PR #2997. 
It should allow migrating also to Storage.StoragePoolType.Filesystem.

rhtyd commented:
@GabrielBrascher I'm not sure what/how you tested it. Did you test using my additional notes: (in a comment shared earlier, now I've added that to the description):

1. To use this feature set the `storage_motion_supported=1` in the hypervisor_capability table for KVM. This is done by default as the feature may not work in some environments, read below.
2. This feature of online storage+VM migration for KVM will only work with CentOS6 and possible Ubuntu as KVM hosts but not with CentOS7 due to:
- https://bugs.centos.org/view.php?id=14026
- https://bugzilla.redhat.com/show_bug.cgi?id=1219541
On CentOS7 the error we see is: `" error: unable to execute QEMU command 'migrate': this feature or command is not currently supported"` (reference https://ask.openstack.org/en/question/94186/live-migration-unable-to-execute-qemu-command-migrate/). Reading through various lists looks like the migrate feature with qemu may be available with paid versions of RHEL-EV but not centos7 however this works with CentOS6.

To test, after enabling storage motion for KVM, you can try to perform online migration of VM with volumes using the `migrateVirtualMachineWithVolume` API.

rhtyd commented:
@nvazquez can you review and address any pending comments?
@GabrielBrascher I looked at the two classes, ideally, we would need a better mechanism to implement hypervisor specific `DataMotionStrategy`. I think due to the way currently it is designed, @nvazquez we should simply either add a new class or work on the `canHandle`

GabrielBrascher commented on engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java:
```diff
@@ -2046,6 +2207,46 @@ private void verifyLiveMigrationMapForKVM(Map<VolumeInfo, DataStore> volumeDataS
+            }
+
+            if (!destStoragePoolVO.isManaged()) {
+                if (destStoragePoolVO.getScope() != ScopeType.CLUSTER) {
```
In the case of local storage migration, the storage pool scope type is of `ScopeType.HOST;` therefore, this is also breaking for local storage migration, throwing the exception `CloudRuntimeException`.

nvazquez replied:
Thanks for pointing this out @GabrielBrascher, added NFS check to fix the bug

nvazquez commented:
@rhtyd @GabrielBrascher I moved the `canHandle` check from `StorageDataMotionStrategy` to `KVMNonManagedStorageStrategy` and fixed one of the issues reported which failed to migrate local storage.
@GabrielBrascher can you describe your failing tests with NFS?

GabrielBrascher commented:
@rhtyd @nvazquez the test was done with Ubuntu and `storage_motion_supported=1` and migrating KVM VM between local storage pools; thus, the the storage pool type of `Storage.StoragePoolType.Filesystem` and storage pool scope of `ScopeType.HOST`.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2584

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
<b>Trillian test result (tid-3365)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 26721 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3365-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 69 look OK, 1 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_vpc_privategw_static_routes | `Failure` | 147.78 | test_privategw_acl.py

rhtyd commented:
On hold - we need to refactor the conflicts to make it work with the similar feature with local storage.

rhtyd commented:
@nvazquez can you rebase against latest master and fix conflicts?

rhtyd commented:
ping @nvazquez

nvazquez commented:
Yes @rhtyd I'll address this tomorrow

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2827

nvazquez commented:
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3632)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 26437 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3632-kvm-centos7.zip
Smoke tests completed. 70 look OK, 0 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---

rhtyd commented:
@nvazquez is this ready for review/merging?

nvazquez commented:
@rhtyd I have rebased master and fixed a small issue to allow NFS live storage migrations. Also verified this is working for CentOS 7 using qemu-kvm-ev. It is now ready for review.

Fix for CentOS 7:
- Create repo file on /etc/yum.repos.d/:
````
[qemu-kvm-rhev]
name=oVirt rebuilds of qemu-kvm-rhev
baseurl=http://resources.ovirt.org/pub/ovirt-3.5/rpm/el7Server/
mirrorlist=http://resources.ovirt.org/pub/yum-repo/mirrorlist-ovirt-3.5-el7Server
enabled=1
skip_if_unavailable=1
gpgcheck=0
````
- `yum install qemu-kvm-common-ev-2.3.0-29.1.el7.x86_64 qemu-kvm-ev-2.3.0-29.1.el7.x86_64 qemu-img-ev-2.3.0-29.1.el7.x86_64`
- Reboot host

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-2854

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2855

nvazquez commented:
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3661)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 28839 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3661-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_direct_download.py
Smoke tests completed. 70 look OK, 1 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_upload_direct_download_certificates | `Failure` | 0.07 | test_direct_download.py

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2856

nvazquez commented:
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3662)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 29564 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3662-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_direct_download.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Smoke tests completed. 69 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_upload_direct_download_certificates | `Failure` | 0.04 | test_direct_download.py
test_02_redundant_VPC_default_routes | `Error` | 0.75 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 317.93 | test_vpc_redundant.py

rhtyd commented:
@nvazquez can you check the `test_02_upload_direct_download_certificates` failure?
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2864

nvazquez commented:
@blueorangutan package

blueorangutan commented:
@nvazquez a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-2865

nvazquez commented:
@blueorangutan test

blueorangutan commented:
@nvazquez a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3669)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 25753 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2983-t3669-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 71 look OK, 0 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---

nvazquez commented:
@rhtyd marvin test failure fixed, now all tests are passing
